tests/storage/datavolume.go

```diff
@@ -281,7 +281,21 @@ var _ = SIGDescribe("DataVolume Integration", func() {
 			if pvCapacitySize.Cmp(pvcRequestSize) > 0 {
 				Expect(getVirtualSize(vmi, dataVolume)).To(Equal((pvcRequestSize.Value())))
 			} else {
-				overheadPercentage, err := strconv.ParseFloat(string(*vmi.Status.VolumeStatus[1].PersistentVolumeClaimInfo.FilesystemOverhead), 64)
+				volumeStatus := v1.VolumeStatus{}
+				Eventually(func() bool {
+					vmi, err = virtClient.VirtualMachineInstance(vmi.Namespace).Get(context.Background(), vmi.Name, metav1.GetOptions{})
+					Expect(err).ToNot(HaveOccurred())
+					for _, volStatus := range vmi.Status.VolumeStatus {
+						if volStatus.Name == "disk0" {
+							volumeStatus = volStatus
+							return true
+						}
+					}
+					return false
```

Comment:
maybe instead of the true/false dance just return the field you're interested in? or the entire volumestatus?

Reply:
I think the code would remain almost the same in terms of length/readability, as we would need to assign it elsewhere even if we return the volume status.

Reply:
Yeah but you're going to have to deal with an obfuscated error if this ever fails ("expected false to be true")

Reply:
Added a message in the `eventually` so we get a more verbose error if it fails.